samples/openapi3/client/petstore/spring-cloud-3-with-optional/src/main/java/org/openapitools/model/Category.java

```diff
@@ -20,12 +20,12 @@
 @Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
 public class Category {
 
-  private Long id;
+  private Optional<Long> id = Optional.empty();
 
-  private String name;
+  private Optional<@Pattern(regexp = "^[a-zA-Z0-9]+[a-zA-Z0-9\\.\\-_]*[a-zA-Z0-9]+$") String> name = Optional.empty();
 
   public Category id(Long id) {
-    this.id = id;
+    this.id = Optional.of(id);
```

Comment:
Should this be `Optional.ofNullable()`? Or is the decision to explicitly disallow setting `null`?

Reply:
Yes that's a good point. 
I would say it should not null, because you should not use id(null), because avoiding null values, but maybe a matter of taste.

Reply:
Can/should there be a `clearId()` method then?

Otherwise, the only way to set this to `null` or remove the value is to do...

```
category.setId(Optional.empty())
```

Maybe we don't need it 🤷 Seems like an edge case anyway

Reply:
I'll probably ask 4 people and get 5 opinions :D

Reply:
I never used a clearXYZ method to handle or clear Optionals, but why not. I would wait for more feedback and then the community maybe can decide which option is better.

Personally i like more set directly `Optional.empty()` because then you see you make something special. 

But it the same issue with JsonNullable you cannot clear/undefined it with the fluent pattern. You need the setter